src/javascript/Switch.js

```diff
@@ -108,23 +114,43 @@ export class Switch {
             this.checkboxBlurred(e);
         }.bind(this), false);
 
-        this.element.addEventListener("click", function(e) {
+        this.element.addEventListener("click", function (e) {
+            if (this.disabled) {
+                e.preventDefault();
+                return
+            }
+
             this.checkboxToggled(e);
         }.bind(this), false);
+
+
+        // Bind changes to the disabled attribute
+        const observer = new MutationObserver((mutations) => {
+            mutations.forEach((mutation) => {
+                // Check the modified attributeName is "disabled"
+                if (mutation.attributeName === "disabled") {
+                    const disabled = !!mutation.target.attributes["disabled"];
+                    this.checkboxDisabled(disabled);
+                }
```

Comment:
NIT: This is also a somewhat stylistic nit, but I prefer if there is a little as possible business logic in the constructor, and would much rather this logic be in some method that is called from here. I know this isn't _strictly_ the constructor, but `bind()` feels like an extension of "setup-like behavior".